src/coprocessor/dag/expr/builtin_miscellaneous.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::net::IpAddr;
+use std::str::FromStr;
+
+use super::{EvalContext, Result, ScalarFunc};
+use coprocessor::codec::Datum;
+
+impl ScalarFunc {
+    pub fn is_ipv4(&self, ctx: &mut EvalContext, row: &[Datum]) -> Result<Option<i64>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        let s = String::from_utf8(input.into_owned())?;
+        if is_given_string_ipv4(&s) {
+            Ok(Some(1))
+        } else {
+            Ok(Some(0))
+        }
+    }
+}
+
+// Useful utility method - should it go to util.rs in coprocessor?
```

Comment:
I think we don't need to move it to the util.rs now, so we can remove this comment. 

Rest LGTM

Reply:
@siddontang thanks for review, I've removed the comments